packages/components/checkbox/src/Checkbox.tsx

```diff
@@ -66,12 +65,45 @@ export const Checkbox = forwardRef<HTMLButtonElement, CheckboxProps>(
           aria-describedby={description}
           aria-invalid={isInvalid}
           onCheckedChange={handleCheckedChange}
+          /**
+           * If the checkbox doesn't have any direct label (children) then we should try to
+           * get an eventual alternative label from FormField.
+           * On last resort, we shouldn't forget to define an aria-label attribute.
+           */
+          aria-labelledby={children ? innerLabelId : field.labelId}
```

Comment:
This is not needed if you are already using `htmlFor`

Reply:
I'm not so sure... Maybe it is related to our checkbox element being a button and not an input, but the `htmlFor` doesn't seem to be enough:
![image](https://github.com/adevinta/spark/assets/66770550/ba6df29c-a6a1-456b-8c49-849476be09bc)